centreon/src/Core/Dashboard/Application/UseCase/PartialUpdateDashboard/PartialUpdateDashboard.php

```diff
@@ -181,6 +186,18 @@ private function updateDashboardAndSave(Dashboard $dashboard, PartialUpdateDashb
 
             $this->writeDashboardRepository->update($updatedDashboard);
 
+            // We want to save the relation between a dashboard and its thumbnail if the thumbnail does exist and is
+            // not the one currently configured on the dashboard
+            if (
+                $updatedDashboard->getThumbnail() !== null
+                && $updatedDashboard->getThumbnail()->getId() !== $dashboard->getThumbnail()?->getId()
+            ) {
+                if (! $this->mediaRepository->existsByPath($updatedDashboard->getThumbnail()->getRelativePath())) {
+                    throw DashboardException::thumbnailNotFound($updatedDashboard->getId());
+                }
```

Comment:
Pure Business question but if the thumbnail does not exist shouldn't we just do nothing ? (so dont add a thumbnail on a dashboard) instead of returning an error

Reply:
This case should not happen when the client is web.
Nothing prevents user to call directly the API to PATCH a dashboard, and it is part of the data validation.
Same logic is applied for instance for the contacts or contact groups shared